src/ctapipe/core/tool.py

```diff
@@ -443,6 +449,19 @@ def run(self, argv=None, raises=False):
                 Provenance().finish_activity(
                     activity_name=self.name, status="interrupted", exit_code=exit_status
                 )
+            except self.custom_exceptions as err:
```

Comment:
I think we should still catch the `SystemExit` with error code != 0 case here, I think this was the cause of the issue for #1881

Reply:
ok, SystemExit treatment is added back